packages/flutter/lib/src/material/tabs.dart

```diff
@@ -570,6 +577,63 @@ class _IndicatorPainter extends CustomPainter {
     _painter!.paint(canvas, _currentRect!.topLeft, configuration);
   }
 
+  /// Applies the stretch effect to the indicator.
+  Rect _applyStretchEffect(Rect rect) {
+    final double index = controller.index.toDouble();
+    final double value = controller.animation!.value;
+
+    // The progress of the animation from 0 to 1
+    late double tabChangeProgress;
+
+    // If we are changing tabs via index, we want to map the progress between 0 and 1
+    if (controller.indexIsChanging) {
+      double progressLeft = (index - value).abs();
+      final int tabsDelta = (controller.index - controller.previousIndex).abs();
+      if (tabsDelta != 0) {
+        progressLeft /= tabsDelta;
+      }
+      tabChangeProgress = 1 - clampDouble(progressLeft, 0.0, 1.0);
+    } else {
+      // Otherwise, the progress is how close we are to the current tab
+      tabChangeProgress = (index - value).abs();
+    }
+
+    // If the animation has finished, there is no need to apply the stretch effect
+    if (tabChangeProgress == 1.0) {
+      return rect;
+    }
```

Comment:
Would `controller.animation.status` tell us earlier if it is completed? We might put this at the top of this function to skip doing work then. Otherwise, should we assert here that our determination that the animation status is finished is true?

Reply:
@Piinks  Goog point! 
We could add a guard with status == completed at the top, but it would only work when changing tabs via tapping. Dragging a TabBarView doesn't trigger an animation "life cycle", so there is never a completed status, making `tabChangeProgress == 1.0` still neccessary.

We could add the early return at the top, and keep the other one as is, but with an assert that it should only happen when `not indexIsChanging` (dragging the TabBarView). What do you think?

Reply:
@Piinks Actually, trying out the proposed assert, it doesn't work. There is a frame where the progress is 1.0 but the status is .forward, not .completed. So maybe just having the early return with the status at the top is ok.

Reply:
Done in 30961c505dbce09a6ceee973475ea6719e0dc93e

Reply:
Thanks for looking into this!